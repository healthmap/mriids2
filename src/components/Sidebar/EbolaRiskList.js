import React from "react";
import { BlockPadded } from "../styled-components/Block";

const EbolaRiskList = () => (
  <BlockPadded>
    <p>
      Countries with the highest relative risk of Ebola spread
    </p>

    <ol>
      <li>Guinea</li>
      <li>Nigeria</li>
      <li>Ivory Coast</li>
      <li>Sierra Leone</li>
      <li>Mali</li>
      <li>Liberia</li>
      <li>Senegal</li>
      <li>Ghana</li>
      <li>Burkina Faso</li>
      <li>Democratic Republic of the Congo</li>
    </ol>
  </BlockPadded>
);

export default EbolaRiskList;
