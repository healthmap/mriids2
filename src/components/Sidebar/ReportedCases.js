import React from "react";
import { BlockPadded } from "../styled-components/Block";
import {
  ReportedCasesParent,
  ReportedCasesColor,
  ReportedCasesLabel,
  ReportedCasesValue,
} from "../styled-components/ReportedCasesStyles";

const ReportedCases = ({
  dateRange,
  diseaseCaseCount = 0,
  projectedCaseCount = 0,
  projection = false,
}) => {
  const titleText = projection ? "Projection" : "Reported Cases";
  const labelText = projection
    ? "Total outbreak projections"
    : "Suspected and confirmed";
  const iconColor = projection ? "#259994" : "#4D73CE";
  return (
    <BlockPadded className="reported-cases-wrapper">
      <p>
        <strong>
          {titleText} from:
          <br />
          {dateRange.from.toDateString()} to {dateRange.to.toDateString()}
        </strong>
      </p>
      <ReportedCasesParent>
        <ReportedCasesLabel>{labelText}</ReportedCasesLabel>
        <ReportedCasesColor style={{ backgroundColor: iconColor }} />
        <ReportedCasesValue>{diseaseCaseCount}</ReportedCasesValue>
      </ReportedCasesParent>
      {projection && (
        <ReportedCasesParent>
          <ReportedCasesLabel>Projected future cases</ReportedCasesLabel>
          <ReportedCasesColor style={{ backgroundColor: "#F2AD33" }} />
          <ReportedCasesValue>{projectedCaseCount}</ReportedCasesValue>
        </ReportedCasesParent>
      )}
    </BlockPadded>
  );
};

export default ReportedCases;
