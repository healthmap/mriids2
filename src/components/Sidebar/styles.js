import styled from "styled-components";

export const SidebarWrapper = styled.div`
  position: fixed;
  top: ${({ isProjectionsBannerDisplayed, theme: { headerHeights } }) =>
    isProjectionsBannerDisplayed
      ? headerHeights.withBanner
      : headerHeights.default};
  left: 0;
  bottom: 0;
  border-right: 1px solid ${(props) => props.theme.borderLightColor};
  width: calc(${(props) => props.theme.sidebarWidth} - 3.2rem);
  padding: 0.8rem 1.6rem 2.4rem;
  background-color: ${(props) => props.theme.background2};
  overflow: auto;
  z-index: 2; /* $z-index--sidebar make variable */
`;
