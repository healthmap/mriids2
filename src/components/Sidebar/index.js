import React from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import {
  changeCountryFilter,
  changeOutbreakFilter,
} from "../../actions/filters";
import Select from "../Select";
import SidebarCount from "./SidebarCount";
import Summary from "./Summary";
import EbolaRiskList from "./EbolaRiskList";
import * as Styled from "./styles";
import {
  SelectCountryWrapper,
  SelectOutbreakWrapper,
} from "../styled-components/SelectWrappers";
import { getFutureProjectionCount } from "../../utils/ebolaDataHelpers";
import { getDiseaseCount } from "../../utils/sidebarDataHelpers";
import CountrySelect from "../CountrySelect";

const Sidebar = ({
  filters,
  ebolaData,
  ebolaDataCombined,
  covidCaseCountData,
  covidDeathCountData,
  changeCountryFilter,
  changeOutbreakFilter,
}) => {
  const changeOutbreak = (selectedValue) => {
    changeOutbreakFilter(selectedValue.target.value);
    // This resets the country filter to 'All' whenever you switch between outbreaks
    changeCountryFilter("All");
  };

  // This is the disease count for the SidebarCount child component
  const diseaseCount = getDiseaseCount(
    ebolaData,
    covidCaseCountData,
    covidDeathCountData,
    filters
  );

  // This is the projected disease count for the SidebarCount child component
  const projectedDiseaseCount = getFutureProjectionCount(
    ebolaData,
    ebolaDataCombined,
    filters
  );

  const showSidebarCount = filters.view === "snapshot";
  const showEbolaSummary = filters.outbreak === "Ebola Outbreak";
  const showEbolaRiskList =
    filters.view === "risk" && filters.outbreak === "Ebola Outbreak";

  return (
    <Styled.SidebarWrapper>
      <SelectCountryWrapper>
        <CountrySelect />
      </SelectCountryWrapper>
      <SelectOutbreakWrapper>
        <Select
          name="outbreak"
          type="outbreak"
          optionsCurrent={["COVID 19"]}
          optionsPast={["Ebola Outbreak"]}
          value={filters.outbreak}
          changeFunction={changeOutbreak}
        />
      </SelectOutbreakWrapper>
      {showSidebarCount && (
        <SidebarCount
          filters={filters}
          diseaseCount={diseaseCount.toLocaleString()}
          projectedDiseaseCount={projectedDiseaseCount.toLocaleString()}
        />
      )}
      {showEbolaSummary && (
        <Summary
          projection={filters.projection}
          dateRange={filters.dateRange}
          country={filters.country}
          diseaseCaseCount={diseaseCount.toLocaleString()}
        />
      )}
      {showEbolaRiskList && <EbolaRiskList />}
    </Styled.SidebarWrapper>
  );
};

const mapStateToProps = (state) => ({
  filters: state.filters,
  ebolaData: state.ebola.ebolaData.data,
  ebolaDataCombined: state.ebola.ebolaDataCombined.data,
  covidCaseCountData: state.covid.caseCounts.data,
  covidDeathCountData: state.covid.deathCounts.data,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      changeCountryFilter,
      changeOutbreakFilter,
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(Sidebar);
