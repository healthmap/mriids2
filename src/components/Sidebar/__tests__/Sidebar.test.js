import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Sidebar from "../index";
import renderer from "react-test-renderer";
import "jest-styled-components";

const mockStore = configureStore([thunk]);

describe("Tests for the connected Sidebar component", () => {
  let store;
  let component;

  const initialState = {
    ebola: {
      ebolaData: {
        isFetching: 0,
        data: {},
        error: {},
      },
      ebolaDataCombined: {
        isFetching: 0,
        data: {},
        error: {},
      },
      riskData: {
        isFetching: 0,
        data: {},
        error: {},
      },
    },
    filters: {
      country: "All",
      outbreak: "Ebola Outbreak",
      view: "snapshot",
      projection: false,
      dateRange: { from: new Date(2014, 9, 1), to: new Date(2016, 1, 20) },
    },
  };

  beforeEach(() => {
    store = mockStore(initialState);

    component = renderer.create(
      <Provider store={store}>
        <Sidebar />
      </Provider>
    );
  });
  test("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
