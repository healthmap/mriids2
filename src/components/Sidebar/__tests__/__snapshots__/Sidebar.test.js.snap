// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the connected Sidebar component with reduxInitialState should render with given state from Redux store 1`] = `
.c2 {
  margin: 2.4rem 1.2rem 0.4rem;
  font-size: 1.2rem;
  line-height: 2;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
  text-transform: uppercase;
  display: block;
  font-weight: 700;
}

.c5.c5 {
  background-color: #fff;
  font-size: 1.4rem;
  line-height: 2;
  padding: 0 1.2rem;
  display: block;
  border-radius: 3px;
  border: 1px solid;
}

.c5.c5.c5.c5:hover {
  border: 1px solid #fff;
  box-shadow: 0 0 0 4px;
}

.c5.c5.c5.c5.Mui-focused {
  border: 1px solid;
}

.c5.c5.c5.c5:hover:before,
.c5.c5:before {
  border: 0px;
}

.c5.c5 .MuiSelect-select:focus {
  background-color: transparent;
}

.c11 {
  padding: 1.3rem;
}

.c11.reported-cases-wrapper {
  border-bottom: 1px solid #ccc;
}

.c12 {
  margin: 0.6rem 0;
}

.c13 {
  margin-bottom: 0.3rem;
}

.c14 {
  display: inline-block;
  font-size: 1.8em;
}

.c0 {
  position: fixed;
  left: 0;
  bottom: 0;
  border-right: 1px solid;
  width: calc( - 3.2rem);
  padding: .8rem 1.6rem 2.4rem;
  overflow: auto;
  z-index: 2;
}

.c4 select {
  color: #fff;
  background-color: #e43b46;
}

.c4 option {
  color: #fff;
  background-color: #e43b46;
}

.c1 select {
  color: #fff;
  background-color: #000;
}

.c1 option {
  color: #fff;
  background-color: #000;
}

.c6 {
  font-size: 1.4rem;
  line-height: 2;
  padding: 0.9rem 1.2rem;
  width: 100%;
  text-align: left;
  background-color: #fff;
  border-radius: 3px;
  border: 1px solid;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
}

.c6:focus {
  background: #fff;
  border: 1px solid;
}

.c6:hover {
  border: 1px solid #fff;
  box-shadow: 0 0 0 4px;
}

.c6:active {
  background-color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6:disabled {
  cursor: auto;
  opacity: 0.65;
}

.c3 .MuiInputBase-root {
  background-color: #fff;
  font-size: 1.4rem;
  padding: 1.2rem !important;
}

.c3 .MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline {
  border-color: #fff;
  box-shadow: 0 0 0 4px;
}

.c3 .Mui-focused .MuiOutlinedInput-notchedOutline {
  border: 1px solid !important;
  box-shadow: 0 0 0 0;
}

.c3 .MuiAutocomplete-input {
  padding: 0px !important;
  height: 2.4rem;
}

.c3 .MuiSvgIcon-root {
  font-size: 2rem;
}

.c3 .MuiAutocomplete-endAdornment {
  top: calc(50% - 12px);
}

.c7 {
  display: block;
}

.c8 {
  margin-left: 1.2rem;
}

.c10 .MuiSvgIcon-root {
  width: 2rem;
  height: 2rem;
}

.c10.c10 {
  padding: 6px;
}

.c9 .MuiTypography-body1 {
  font-size: 1.4rem;
  line-height: 2;
  font-weight: 600;
  margin-left: 0.2rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Location
    </label>
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root c3 MuiAutocomplete-hasPopupIcon"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            disabled={false}
            id="country-select"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-2"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c4"
  >
    <label
      className="c2"
    >
      outbreak
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline c5"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="mui-component-select-outbreak"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="mui-component-select-outbreak"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Ebola Outbreak
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        name="outbreak"
        onAnimationStart={[Function]}
        onChange={[Function]}
        tabIndex={-1}
        value="Ebola Outbreak"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <label
    className="c2"
  >
    Timespan
  </label>
  <div>
    <button
      className="c6"
      onClick={[Function]}
    >
      Oct 1, 2014
       -
       
      Feb 20, 2016
    </button>
  </div>
  <div
    className="c7"
  >
    <fieldset
      className="MuiFormControl-root"
    >
      <label
        className="c2"
      >
        Data
      </label>
      <div
        aria-label="data"
        className="MuiFormGroup-root c8"
        role="radiogroup"
      >
        <label
          className="MuiFormControlLabel-root c9"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root MuiRadio-colorSecondary c10 PrivateSwitchBase-checked-6 Mui-checked MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={true}
                className="PrivateSwitchBase-input-8"
                disabled={false}
                name="data"
                onChange={[Function]}
                type="radio"
                value="cases"
              />
              <div
                className="PrivateRadioButtonIcon-root-9 PrivateRadioButtonIcon-checked-11"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Cases
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root c9"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root MuiRadio-colorSecondary c10 MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-8"
                disabled={false}
                name="data"
                onChange={[Function]}
                type="radio"
                value="projected cases"
              />
              <div
                className="PrivateRadioButtonIcon-root-9"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Projected Cases
          </span>
        </label>
      </div>
    </fieldset>
  </div>
  <div
    className="c11 reported-cases-wrapper"
  >
    <p>
      <strong>
        Reported cases
         from:
        <br />
        Wed Oct 01 2014
         to
         
        Sat Feb 20 2016
      </strong>
    </p>
    <div
      className="c12"
    >
      <div
        className="c13"
      >
        Suspected and confirmed
      </div>
      <div
        className="c14"
      >
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests for the connected Sidebar component with riskViewState should render with given state from Redux store 1`] = `
.c2 {
  margin: 2.4rem 1.2rem 0.4rem;
  font-size: 1.2rem;
  line-height: 2;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
  text-transform: uppercase;
  display: block;
  font-weight: 700;
}

.c5.c5 {
  background-color: #fff;
  font-size: 1.4rem;
  line-height: 2;
  padding: 0 1.2rem;
  display: block;
  border-radius: 3px;
  border: 1px solid;
}

.c5.c5.c5.c5:hover {
  border: 1px solid #fff;
  box-shadow: 0 0 0 4px;
}

.c5.c5.c5.c5.Mui-focused {
  border: 1px solid;
}

.c5.c5.c5.c5:hover:before,
.c5.c5:before {
  border: 0px;
}

.c5.c5 .MuiSelect-select:focus {
  background-color: transparent;
}

.c11 {
  padding: 1.3rem;
}

.c11.reported-cases-wrapper {
  border-bottom: 1px solid #ccc;
}

.c0 {
  position: fixed;
  left: 0;
  bottom: 0;
  border-right: 1px solid;
  width: calc( - 3.2rem);
  padding: .8rem 1.6rem 2.4rem;
  overflow: auto;
  z-index: 2;
}

.c4 select {
  color: #fff;
  background-color: #e43b46;
}

.c4 option {
  color: #fff;
  background-color: #e43b46;
}

.c1 select {
  color: #fff;
  background-color: #000;
}

.c1 option {
  color: #fff;
  background-color: #000;
}

.c6 {
  font-size: 1.4rem;
  line-height: 2;
  padding: 0.9rem 1.2rem;
  width: 100%;
  text-align: left;
  background-color: #fff;
  border-radius: 3px;
  border: 1px solid;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
}

.c6:focus {
  background: #fff;
  border: 1px solid;
}

.c6:hover {
  border: 1px solid #fff;
  box-shadow: 0 0 0 4px;
}

.c6:active {
  background-color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6:disabled {
  cursor: auto;
  opacity: 0.65;
}

.c3 .MuiInputBase-root {
  background-color: #fff;
  font-size: 1.4rem;
  padding: 1.2rem !important;
}

.c3 .MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline {
  border-color: #fff;
  box-shadow: 0 0 0 4px;
}

.c3 .Mui-focused .MuiOutlinedInput-notchedOutline {
  border: 1px solid !important;
  box-shadow: 0 0 0 0;
}

.c3 .MuiAutocomplete-input {
  padding: 0px !important;
  height: 2.4rem;
}

.c3 .MuiSvgIcon-root {
  font-size: 2rem;
}

.c3 .MuiAutocomplete-endAdornment {
  top: calc(50% - 12px);
}

.c7 {
  display: block;
}

.c8 {
  margin-left: 1.2rem;
}

.c10 .MuiSvgIcon-root {
  width: 2rem;
  height: 2rem;
}

.c10.c10 {
  padding: 6px;
}

.c9 .MuiTypography-body1 {
  font-size: 1.4rem;
  line-height: 2;
  font-weight: 600;
  margin-left: 0.2rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Location
    </label>
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root c3 MuiAutocomplete-hasPopupIcon"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            disabled={false}
            id="country-select"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            required={false}
            spellCheck="false"
            type="text"
            value="All"
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-2"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c4"
  >
    <label
      className="c2"
    >
      outbreak
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline c5"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="mui-component-select-outbreak"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="mui-component-select-outbreak"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Ebola Outbreak
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        name="outbreak"
        onAnimationStart={[Function]}
        onChange={[Function]}
        tabIndex={-1}
        value="Ebola Outbreak"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <label
    className="c2"
  >
    Timespan
  </label>
  <div>
    <button
      className="c6"
      onClick={[Function]}
    >
      Oct 1, 2014
       -
       
      Feb 20, 2016
    </button>
  </div>
  <div
    className="c7"
  >
    <fieldset
      className="MuiFormControl-root"
    >
      <label
        className="c2"
      >
        Data
      </label>
      <div
        aria-label="data"
        className="MuiFormGroup-root c8"
        role="radiogroup"
      >
        <label
          className="MuiFormControlLabel-root c9"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root MuiRadio-colorSecondary c10 PrivateSwitchBase-checked-6 Mui-checked MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={true}
                className="PrivateSwitchBase-input-8"
                disabled={false}
                name="data"
                onChange={[Function]}
                type="radio"
                value="cases"
              />
              <div
                className="PrivateRadioButtonIcon-root-9 PrivateRadioButtonIcon-checked-11"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Cases
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root c9"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root MuiRadio-colorSecondary c10 MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-8"
                disabled={false}
                name="data"
                onChange={[Function]}
                type="radio"
                value="projected cases"
              />
              <div
                className="PrivateRadioButtonIcon-root-9"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Projected Cases
          </span>
        </label>
      </div>
    </fieldset>
  </div>
  <div
    className="c11"
  >
    <strong>
      <p>
        The top 10 countries with the highest relative risk of Ebola spread are:
      </p>
    </strong>
    <ol>
      <li>
        Guinea
      </li>
      <li>
        Nigeria
      </li>
      <li>
        Ivory Coast
      </li>
      <li>
        Sierra Leone
      </li>
      <li>
        Mali
      </li>
      <li>
        Liberia
      </li>
      <li>
        Senegal
      </li>
      <li>
        Ghana
      </li>
      <li>
        Burkina Faso
      </li>
      <li>
        Democratic Republic of the Congo
      </li>
    </ol>
  </div>
</div>
`;
